src/modelinstance.cpp

```diff
@@ -351,16 +351,47 @@ static Status applyLayoutConfiguration(const ModelConfig& config, std::shared_pt
     return StatusCode::OK;
 }
 
-const std::string RT_INFO_KEY{"model_info"};
+ov::AnyMap ModelInstance::getRTInfo(std::vector<std::string> path) {
+    ov::AnyMap anyMap, rtMap;
+    try {
+        rtMap = model->get_rt_info<ov::AnyMap>(path);
+    } catch (const ov::Exception& e) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Failed to get RTInfo; error:{}", e.what());
+        return anyMap;
+    } catch (...) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Failed to get RTInfo for path; unknown error");
+        return anyMap;
+    }
+    std::string type_name("St10shared_ptrIN2ov4MetaEE");
```

Comment:
Doesn't it change when OV is recompiled?

Reply:
https://en.cppreference.com/w/cpp/types/type_info/name

> Returns an implementation defined null-terminated character string containing the name of the type. No guarantees are given; in particular, the returned string can be identical for several types and change between invocations of the same program.

What you could try forward declare ov::Meta and use typeid(*value) == typeid(std::shared_ptr<ov::Meta>), but still it would be good to check on different compiler versions and potentially Windows as well.

Reply:
Has this been answered? Is this line safe?

Reply:
we can't refer to ov::Meta because it is ov private class. It will not compile. We would need to confirm if on windows and arm the same name will be reported.

Reply:
Class identifier should suffice here for typeid so you could do sth like this:
namespace ov {
struct Meta; // pure fwd declaration
}
(typeid(std::shared_ptr<ov::Meta>) == value.type_info())

@mzegla @dkalinowski @dtrawins